src/Config/Converters/EntityRestOptionsConverter.cs

```diff
@@ -0,0 +1,113 @@
+    {
+        if (reader.TokenType == JsonTokenType.StartObject)
+        {
+            EntityRestOptions restOptions = new(Methods: Array.Empty<SupportedHttpVerb>(), Path: null, Enabled: true);
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    break;
+                }
+
+                string? propertyName = reader.GetString();
+
+                switch (propertyName)
+                {
+                    case "path":
+                    {
+                        reader.Read();
+
+                        if (reader.TokenType == JsonTokenType.String)
+                        {
+                            restOptions = restOptions with { Path = reader.DeserializeString() };
+                            break;
+                        }
+
+                        if (reader.TokenType == JsonTokenType.True || reader.TokenType == JsonTokenType.False)
+                        {
+                            restOptions = restOptions with { Enabled = reader.GetBoolean() };
+                            break;
+                        }
+
+                        break;
+                    }
+
+                    case "methods":
+                    {
+                        List<SupportedHttpVerb> methods = new();
+                        while (reader.Read())
+                        {
+                            if (reader.TokenType == JsonTokenType.StartArray)
+                            {
+                                continue;
+                            }
+
+                            if (reader.TokenType == JsonTokenType.EndArray)
+                            {
+                                break;
+                            }
+
+                            methods.Add(EnumExtensions.Deserialize<SupportedHttpVerb>(reader.DeserializeString()!));
+                        }
+
+                        restOptions = restOptions with { Methods = methods.ToArray() };
+                        break;
+                    }
+
+                    case "enabled":
+                    {
+                        reader.Read();
+                        restOptions = restOptions with { Enabled = reader.GetBoolean() };
+                        break;
+                    }
+
+                    default:
+                        throw new JsonException($"Unexpected property {propertyName}");
+                }
+            }
+
+            return restOptions;
+        }
+
+        if (reader.TokenType == JsonTokenType.String)
+        {
+            return new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS, reader.DeserializeString(), true);
+        }
+
+        if (reader.TokenType == JsonTokenType.True || reader.TokenType == JsonTokenType.False)
+        {
+            return new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS, null, reader.GetBoolean());
```

Comment:
if its JsonTokenType.False i.e. Enabled = False, we should nullify the methods supported.

For JsonTokenType.True, shouldn't the Path be set to the EntityName by default? I think while serializing we do that but why not while deserializing?